samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```
@@ -130,29 +130,31 @@ foreach(GROUP BASE NOTIFICATIONS ALERTS STATELISTENER SPEECHDATA MUSICDATA CUSTO
         get_filename_component(FILE_BASE ${file} NAME_WE)
         list(APPEND PROTOC_INC_DIRS "-I.")
         list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
-        list(APPEND ${GROUP}_PROTO_BYPRODUCTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+        list(APPEND ${GROUP}_PROTO_OUTPUTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+        list(APPEND ${GROUP}_PROTO_DEPENDENCIES "${file}")
     endforeach()
 
-    # Populate list of compilation commands for protoc compiler.
-    unset(PROTOC_CMD_LIST)
+    # Each .pb.c file to be compiled by protoc gets its own CMake custom command,
+    # with dependencies to all other .proto files in its group and the base files.
     foreach(file ${${GROUP}_PROTO_FILES})
         get_filename_component(FILE_PATH ${file} DIRECTORY)
         get_filename_component(FILE_NAME ${file} NAME)
+        get_filename_component(FILE_BASE ${file} NAME_WE)
```

```suggestion
        get_filename_component(FILE_BASE_NAME ${file} NAME_WE)
```
to make it a bit clearer, as you also are using `BASE` as the name of a group.

Will also be a similar to other parts of CMake, as:
```
$<TARGET_FILE_BASE_NAME:tgt>
    Base name of tgt, i.e. $<TARGET_FILE_NAME:tgt> without prefix and suffix. 
    For example, if the tgt filename is libbase.so, the base name is base.
```

---

Updating name as suggested